rl_engine/moe/contract.py

```
@@ -0,0 +1,224 @@
+    w_fc2: torch.Tensor  # BF16 [hidden, ffn] frozen
+    placement: str = "replicated"
+    schema_version: str = SCHEMA_VERSION
+    numeric_profile: str = ORACLE_PROFILE
+    metadata: dict[str, Any] = field(default_factory=dict)
+
+    def validate(self) -> None:
+        if self.x.dtype != torch.bfloat16:
+            raise TypeError(f"x must be BF16, got {self.x.dtype}")
+        if self.w_fc1.dtype != torch.bfloat16 or self.w_fc2.dtype != torch.bfloat16:
+            raise TypeError("shared weights must be BF16 in the v1 contract")
+        t, hidden = self.x.shape
```

Ruff (0.16.3)

[warning] Unpacked variable `t` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)